src/common/random_generator.cu

```diff
@@ -59,6 +59,17 @@ void RandGenerator<gpu, float>::Seed(mshadow::Stream<gpu> *s, uint32_t seed) {
   s->Wait();
 }
 
+template<>
+void RandGenerator<gpu, float>::AllocState(RandGenerator<gpu> *inst) {
```

Comment:
doesnt this remove support for other DTypes like half_t ?

Reply:
No, the random generator always generates floats between 0 and 1, we scale them as needed in wherever the random generator is used